File: src/main/java/pl/com/tegess/controller/admin/writecontroller/AdminWriteController.java
package pl.com.tegess.controller.admin.writecontroller;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import pl.com.tegess.controller.admin.request.AdminWriteData;
import pl.com.tegess.domain.admin.Admin;
import pl.com.tegess.domain.admin.AdminRepository;

@RestController
@Component
public class AdminWriteController {

    @Autowired
    AdminRepository repository;

    @RequestMapping(value = "/api/admins", method = RequestMethod.POST)
    public void create(@RequestBody AdminWriteData adminWriteData) {
        Admin admin = new Admin(adminWriteData.getUsername(), adminWriteData.getPassword());
        repository.insert(admin);
    }
}

File: src/main/scala/com/tegess/controller/admin/AdminWriteController.scala
package com.tegess.controller.admin

import com.tegess.controller.admin.AdminWriteController.AdminWriteData
import com.tegess.domain.admin.Admin
import com.tegess.persistance.service.admin.AdminService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.{RequestBody, RequestMapping, RequestMethod, RestController}

@RestController
@Component
class AdminWriteController {

  @Autowired
  var adminService: AdminService = _

  @RequestMapping(value=Array("/api/install"), method=Array(RequestMethod.POST))
  def install(@RequestBody adminWriteData: AdminWriteData) = {
    val admin: Admin = Admin(adminWriteData.username, adminWriteData.password, List())
    adminService.save(admin)
  }

}

object AdminWriteController {
  case class AdminWriteData(username: String, password: String)
}
